tests/foreman/ui/test_contentview.py

```diff
@@ -2297,3 +2298,50 @@ def test_positive_publish_promote_with_rh_ostree_and_other(self):
                 self.content_views.wait_until_element(
                     common_locators['alert.success_sub_form'])
             )
+
+    @run_in_one_thread
+    @run_only_on('sat')
+    @tier2
+    def test_positive_promote_CV_with_custom_user_role_and_filters(self):
+        """Publish and promote cv with user with custom role and filter
+
+        @id: a07fe3df-8645-4a0c-8c56-3f8314ae4878
+
+        @Assert: CV should be published and promoted successfully
+
+        @CaseLevel: Integration
+        """
+        role = entities.Role().create()
+        for perm in CUST_PERMS_LIST:
+            permissions = entities.Permission(name=perm).search()
+            entities.Filter(permission=permissions, role=role).create()
```

Comment:
Tried to do this failed, 
Rechecked the filters and permissions association , So apparently we will requires more than one filter , as filters are grouped by resource type , one filter can have permissions from only one resource type.

`  "error": {"id":null,"errors":{"permissions":["Permissions must be of same resource type"]},"full_messages":["Permissions Permissions must be of same resource type"]}
}`

The custom list of permissions are from across various resource_types